--- a/node_modules/react-joyride/dist/index.mjs
+++ b/node_modules/react-joyride/dist/index.mjs
@@ -1,7 +1,7 @@
 import e, { Component as t, forwardRef as n, PureComponent as r, Children as o, cloneElement as i, useState as s, useMemo as a, useEffect as c, useLayoutEffect as l, useCallback as u } from "react";
-import { findDOMNode as d, unmountComponentAtNode as m, unstable_renderSubtreeIntoContainer as p } from "react-dom";
+import { findDOMNode as d } from "react-dom";
+import { createRoot } from "react-dom/client";
 import h from "deep-diff";
 import { computePosition as f } from "@floating-ui/dom";
 import b from "react-floater";
@@ -27,11 +27,15 @@
   renderSubtree(e) {
     const { isTourRunning: t } = this.props;
     this.unmountSubtree();
-    const n = /* @__PURE__ */ e;
-    t && p(this, n, this.node);
+    if (t) {
+      if (!this.root) {
+        this.root = createRoot(this.node);
+      }
+      this.root.render(e);
+    }
   }
   unmountSubtree() {
-    m(this.node);
+    this.root?.unmount();
   }
   render() {
     return null;