patch-package
--- a/node_modules/react-joyride/dist/index.mjs
+++ b/node_modules/react-joyride/dist/index.mjs
@@ -1,6 +1,6 @@
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
 import React, { useReducer, useEffect, useLayoutEffect, useCallback, useMemo, useRef } from 'react';
-import { Animate } from 'react-simple-animate';
+import { Animate } from 'react-simple-animate/dist/index.mjs';
 import scroll from 'scroll';
 import { computePosition, offset, flip, shift, arrow } from '@floating-ui/dom';
 import { useTour } from '@reactour/tour';
@@ -512,12 +512,14 @@
 
     if (portal) {
       this.portal = portal;
-      document.body.appendChild(this.portal);
     }
 
-    this.renderSubtree();
+    document.body.appendChild(this.portal);
+    this.root = createRoot(this.portal);
+    this.renderSubtree(this.root);
   }
 
+  root;
   componentDidUpdate() {
     this.renderSubtree();
   }
@@ -525,27 +527,16 @@
   componentWillUnmount() {
     if (!this.portal || !this.root) {
       return;
-    }
-
-    ReactDOM.unmountComponentAtNode(this.portal);
-    document.body.removeChild(this.portal);
+    }    
+    this.root.unmount();
+    document.body.removeChild(this.portal)
   }
 
-  renderSubtree() {
-    const { children } = this.props;
-
-    if (!this.portal) {
-      return;
+  renderSubtree(root) {
+    const { root: _root, portal, props } = this;
+    if (!_root || !portal) {
+      return null;
     }
-
-    // eslint-disable-next-line react/no-render-return-value
-    return ReactDOM.unstable_renderSubtreeIntoContainer(
-      this,
-      children,
-      this.portal,
-    );
+    return _root.render(props.children);
   }
 
   render() {
